Server/JokesOnYou.Web.Api/Repositories/UserRepository.cs

```diff
@@ -35,16 +50,64 @@ public async Task<User> CreateUserAsync(UserRegisterDTO userRegisterDTO)
                     message.AppendLine(error.Description);
                 }
 
-                throw new UserRegisterException(message.ToString());
+                throw new UserRegisterException(message.ToString(), new ArgumentException());
             }
         }
 
         public Task DeleteUserAsync(User user) => _userManager.DeleteAsync(user);
 
-        public Task<User> GetUserAsync(string id) => _userManager.FindByIdAsync(id);
+        public async Task<UserReplyDTO> GetUserReplyAsync(string id)
+        {
+            //Possibilities 
+            //var user = await _userManager.FindByIdAsync(id);
+            //var user = await _userManager.Users.FirstOrDefaultAsync(u => u.Id == id);
+            var user = await _userManager.FindByIdAsync(id);
 
-        public Task<User> GetUserByEmail(string email) => _userManager.FindByEmailAsync(email);
+            return _mapper.Map<UserReplyDTO>(user);
+        }
+        public async Task<User> GetUserAsync(string id)
+        {
+            //Possibilities 
+            //var user = await _userManager.FindByIdAsync(id);
+            //var user = await _userManager.Users.FirstOrDefaultAsync(u => u.Id == id);
+            return await _userManager.FindByIdAsync(id);
+        }
+        public Task<User> GetUserByEmail(string email)
+        {
+            return _userManager.FindByEmailAsync(email);
+        }
 
-        public async Task<IEnumerable<User>> GetUsersAsync() => await _userManager.Users.ToListAsync();
+        public async Task<IEnumerable<User>> GetUsersAsync()
+        {
+            //Such a waste of bits
+            return await _userManager.Users.ToListAsync();
```

Comment:
Yes you are right ( and Like Chris told me there is a better way. using mapper.
See this example:
```
public async Task<IEnumerable<JokeReplyDto>> GetAllJokeDtosAsync()
        {
            var jokeDtos = await _context.Jokes.ProjectTo<JokeReplyDto>(_mapper.ConfigurationProvider).ToListAsync();

            return jokeDtos;
        }
```

Reply:
I do the mapping in UserService.cs? Should i move it to the Repository instead?

Reply:
Yes the other way seems better. I didn't know about it untill Chris told me.

Reply:
use the .ProjectTo method so you don't get all the useless info from the database.